mempool/cache.go

```diff
@@ -16,16 +16,16 @@ type TxCache interface {
 	// Reset resets the cache to an empty state.
 	Reset()
 
-	// Push adds the given raw transaction to the cache and returns true if it was
+	// Push adds the given tx key to the cache and returns true if it was
 	// newly added. Otherwise, it returns false.
-	Push(tx types.Tx) bool
+	Push(tx types.TxKey) bool
 
-	// Remove removes the given raw transaction from the cache.
-	Remove(tx types.Tx)
+	// Remove removes the given tx key from the cache.
+	Remove(tx types.TxKey)
 
-	// Has reports whether tx is present in the cache. Checking for presence is
+	// Has reports whether tx key is present in the cache. Checking for presence is
 	// not treated as an access of the value.
-	Has(tx types.Tx) bool
+	Has(tx types.TxKey) bool
```

Comment:
It's not necessary in the sense that it can be achieved without breaking it however we're needlessly computing the hash multiple times and this isn't an interface that we expect users to use. I don't think users can even use it in the sense that the mempool constructor doesn't accept a TxCache interface so this interface would be better of being private (and used just for testing)

Reply:
Fair enough. Given the interface is public this PR still seems breaking so we should consider adding `!` to the PR title. IMO we could've split this into two separate PRs:
1. Add the feature and don't break the interface.
2. A separate PR for breaking the interface to reduce hashing if that's really a problem.

Since this PR targets celestia-core and we don't know how to release breaking changes of this repo, I'm concerned this PR won't get released from `main` anytime soon. Also if it's technically API breaking, I don't think we're supposed to backport v0.34.x-celestia?

If we split this into a non-breaking change and a breaking change, we can easily backport the non-breaking change.

Reply:
It's a bit silly that this has to be breaking. As I alluded to, you can't actually use the interface for anything (you don't get to choose what cache the mempool uses) and it should've been private (if we had wanted it for testing purposes). Since this is only performative and we want this feature in v2 not v3, @ninabarbakadze can you revert these changes to make it non-breaking. If we want we can revisit this later on

Reply:
reverting the changes